arangod/Transaction/Manager.cpp

```diff
@@ -723,5 +746,71 @@ bool Manager::abortManagedTrx(std::function<bool(TransactionState const&)> cb) {
   return !toAbort.empty();
 }
 
+void Manager::toVelocyPack(VPackBuilder& builder, std::string const& database, std::string const& username, bool fanout) const {
+  TRI_ASSERT(!builder.isClosed());
+  
+  if (fanout) {
+    TRI_ASSERT(ServerState::instance()->isCoordinator());
+    auto ci = ClusterInfo::instance();
+    if (ci == nullptr) {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
+    }
+      
+    std::shared_ptr<ClusterComm> cc = ClusterComm::instance();
+    if (cc == nullptr) {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
+    }
+
+    for (auto const& coordinator : ci->getCurrentCoordinators()) {
+      if (coordinator == ServerState::instance()->getId()) {
+        // ourselves!
+        continue;
+      }
+
+      auto auth = AuthenticationFeature::instance();
+      std::unordered_map<std::string, std::string> headers;
+      if (auth != nullptr && auth->isActive()) {
+        // when in superuser mode, username is empty
+        //  in this case ClusterComm will add the default superuser token
+        if (!username.empty()) {
+          VPackBuilder builder;
+          {
+            VPackObjectBuilder payload{&builder};
+            payload->add("preferred_username", VPackValue(username));
+          }
+          VPackSlice slice = builder.slice();
+          headers.emplace(StaticStrings::Authorization,
+                          "bearer " + auth->tokenCache().generateJwt(slice));
+        }
+      }
+  
+      double const timeout = 5.0;
+      
+      auto comres = 
```

Comment:
don't we want to get away from these kind of blocking APIs ? How about we revive the networking v2 PR ?